### PR TITLE
Fips ltsr

### PR DESCRIPTION
**Pull request by piotr-kotara-ibm**

### CODE REVIEW

File: controllers/ibmlicensing_controller.go [hunk 1]
```diff
@@ -377,64 +383,16 @@ func (r *IBMLicensingReconciler) reconcileCertificateSecrets(instance *operatorv
...
-				return res.UpdateResource(&reqLogger, r.Client, secret, ocpExternalCertSecret)
-			}
+		namespacedName = types.NamespacedName{Namespace: instance.Spec.InstanceNamespace, Name: service.LicenseServiceExternalCertName}
+		hostname = route.Spec.Host
+	} else {
+		r.Log.Info("Reconciling internal certificate")
 
-		}
+		namespacedName = types.NamespacedName{Namespace: instance.Spec.InstanceNamespace, Name: service.LicenseServiceInternalCertName}
+		hostname = fmt.Sprintf("%s-%s.svc", service.GetResourceName(instance), instance.Spec.InstanceNamespace)
```
Comment (piotr-kotara-ibm) [inline controllers/ibmlicensing_controller.go:392]: possibly need to add second hostname (svc.local) like ocp

File: controllers/ibmlicensing_controller.go [hunk 2]
```diff
@@ -778,3 +736,56 @@ func (r *IBMLicensingReconciler) controllerStatus(instance *operatorv1alpha1.IBM
...
+	if err != nil {
+		r.Log.Error(err, "Improper x509 certificate in secret")
+		regenerateCertificate = true
+	}
+	// if certificate is expired
+	if cert.NotAfter.Before(time.Now()) {
+		r.Log.Info("Self signed certificate has expired.")
+		regenerateCertificate = true
+	}
+	// if certificate is not issued to the route host
```
Comment (piotr-kotara-ibm) [inline controllers/ibmlicensing_controller.go:774]:
```suggestion
	// if certificate is not issued to proper host
```

File: controllers/resources/service/containers.go [hunk 3]
```diff
@@ -54,18 +54,9 @@ func getLicensingEnvironmentVariables(spec operatorv1alpha1.IBMLicensingSpec) []
...
-			certSource = operatorv1alpha1.OcpCertsSource
-		} else {
-			// on K8s always always generate self-signed certificate in operand
-			certSource = operatorv1alpha1.SelfSignedCertsSource
-		}
-
 		environmentVariables = append(environmentVariables, corev1.EnvVar{
 			Name:  "HTTPS_CERTS_SOURCE",
-			Value: string(certSource),
+			Value: string(operatorv1alpha1.OcpCertsSource),
```
Comment (piotr-kotara-ibm) [inline controllers/resources/service/containers.go:59]: Remove HTTPS_CERTS_SOURCE env completely from operand

File: controllers/ibmlicensing_controller.go [hunk 4]
```diff
@@ -365,10 +365,16 @@ func (r *IBMLicensingReconciler) reconcileDeployment(instance *operatorv1alpha1.
 
 func (r *IBMLicensingReconciler) reconcileCertificateSecrets(instance *operatorv1alpha1.IBMLicensing) (reconcile.Result, error) {
 	// for backward compatibility, we treat the "ocp" HTTPSCertsSource same as "self-signed"
-	if res.IsRouteAPI && instance.Spec.IsRouteEnabled() && instance.Spec.HTTPSCertsSource != "custom" {
-		ocpExternalCertSecret := &corev1.Secret{}
+	if instance.Spec.HTTPSCertsSource == "custom" {
```
Comment (piotr-kotara-ibm) [inline controllers/ibmlicensing_controller.go:368]: we need to change logic to always reconcile internal certificates

File: controllers/ibmlicensing_controller.go [hunk 5]
```diff
@@ -778,3 +736,56 @@ func (r *IBMLicensingReconciler) controllerStatus(instance *operatorv1alpha1.IBM
...
+		r.Log.Info("Self signed certificate has expired.")
+		regenerateCertificate = true
+	}
+	// if certificate is not issued to the route host
+	if err := cert.VerifyHostname(hostname); err != nil {
+		r.Log.Info("Certificate not issued to a propper hostname.")
+		regenerateCertificate = true
+	}
+
+	if regenerateCertificate {
```
Comment (piotr-kotara-ibm) [inline controllers/ibmlicensing_controller.go:780]: we need mechanism for restarting pod here (internal certificate rotation)

**Review by SzymonKowalczyk — APPROVED** [review]

**Review by adamdyszy — APPROVED** [review]

**Pull request merged**